app/uk/gov/hmrc/agentsubscriptionfrontend/GuiceModule.scala

```diff
@@ -35,6 +36,45 @@ class GuiceModule(environment: Environment, configuration: Configuration) extend
   override protected lazy val mode: Mode = environment.mode
   override protected lazy val runModeConfiguration: Configuration = configuration
 
+  trait ConfigPropertyType[A] {
+    def bindConfigProperty(clazz: Class[A])(propertyName: String): ScopedBindingBuilder
+  }
+
+  object ConfigPropertyType {
+    implicit val stringConfigProperty = new ConfigPropertyType[String] {
+      def bindConfigProperty(clazz: Class[String])(propertyName: String): ScopedBindingBuilder =
+        bind(clazz).annotatedWith(Names.named(s"$propertyName")).toProvider(new StringConfigPropertyProvider(propertyName))
+
+      private class StringConfigPropertyProvider(propertyName: String) extends Provider[String] {
```

Comment:
The limitation comes from Typesafe's Config library not supporting truly generic getters. IMHO current state of property providers is a balance between verbosity vs readability.